### PR TITLE
Patches of Hacker‘s morph

### PR DESCRIPTION
**Pull request by ghost**
Main modifications:

- Add controls to GuiTransformations that can rotate with respect to the parent axis
- Keyframe element scales with the cursor position
- GuiModelRenderer adds fields to fit Blockbuster's immersive editor (hideModel and fullScreen)
- GuiModelRenderer adds parameters for custom entity pitch, yawHead, yawBody
- GuiModelRenderer can use flight mode by holding down the left mouse button
- Add VertexBuilder to be compatible with Optifine (mainly used to provide tangent)

All Mods: use UTF-8 encoding to read and save files

### CODE REVIEW

**Comment by mchorse** [comment]
Hey, thanks for all the PRs, but heads up that I'm going to release Mappet rc3 first before looking over these changes.

**Comment by ghost** [comment]
Got it. Before you come back to check out these changes, I may continue create some commits to fix bugs in my code or write some little features that pop into my head. Hope this doesn't disturb your break.

**Comment by mchorse** [comment]
> Got it. Before you come back to check out these changes, I may continue create some commits to fix bugs in my code or write some little features that pop into my head. Hope this doesn't disturb your break.

Yeah, that's fine! Let me know when you'll finish it so I wouldn't merge this pull request before you're finished, though.

**Comment by ghost** [comment]
I just finished my last thought half a day ago and now there's nothing left in my brain XD
Well, actually I'm not very sure if now is the right time to start the merge. Even though I've fixed many of my bugs, I'm still worried that there would be more. I guess I'm a bit nervous about the moment that will finally come.

File: src/main/java/mchorse/mclib/ClientProxy.java [hunk 1]
```diff
@@ -22,13 +23,17 @@
 @SideOnly(Side.CLIENT)
 public class ClientProxy extends CommonProxy
 {
+    public static KeybindConfig keybinds;
```
Comment (mchorse) [inline src/main/java/mchorse/mclib/ClientProxy.java:26]: At the moment I'm ~~busy~~ lazy to take a deep dive into this, but if you could summarize what this commit is about? 👀

Comment (ghost) [reply src/main/java/mchorse/mclib/ClientProxy.java:26]: This commit adds a new config page that allows you to configure hotkeys for each interface in McLib and its submods.
I tried to have each Keybind object add itself to ClientProxy.keybinds at construct time, but, I have to say this really isn't the most appropriate way to do it.

**Comment by ghost** [comment]
Well, I was wrong. After a week of testing, it seems I still have work to do, and I apologize for that.
I hope this doesn't interfere with your work schedule, and I'll let you know when I've finished these things.

File: src/main/java/mchorse/mclib/ClientProxy.java (hunk 1 above)
Comment (mchorse) [reply src/main/java/mchorse/mclib/ClientProxy.java:26]: > This commit adds a new config page that allows you to configure hotkeys for each interface in McLib and its submods.
> I tried to have each Keybind object add itself to ClientProxy.keybinds at construct time, but, I have to say this really isn't the most appropriate way to do it.

Wow! That’s really cool! 😮

**Comment by mchorse** [comment]
> Well, I was wrong. After a week of testing, it seems I still have work to do, and I apologize for that.
> I hope this doesn't interfere with your work schedule, and I'll let you know when I've finished these things.

Yeah, that’s totally cool, there is no need to rush. I can accept PR any time. 👍

**Pull request merged**